### PR TITLE
V3 group calls

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/calling/CallingService.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+'use strict';
+
+window.z = window.z || {};
+window.z.calling = z.calling || {};
+
+z.calling.CallingService = class CallingService {
+  /**
+   * Construct an new CallingService.
+   * @param {z.client.Client} client - Local client entity
+   * @returns {CallingService} The new CallingService
```
Comment (bennycode) [inline app/script/calling/CallingService.js]: `z.calling.CallingService`

File: app/script/calling/entities/ECallMessage.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+};
+
+z.calling.entities.ECallMessage = class ECallMessage {
+  /**
+   * Construct a new e-call message entity.
+   *
+   * @param {z.calling.enum.E_CALL_MESSAGE_TYPE} type - Type of e-call message
+   * @param {boolean} [response=false] - Is message a response, defaults to false
+   * @param {string} session_id - Optional session ID
+   * @returns {ECallMessage} The new e-call message entity
```
Comment (bennycode) [inline app/script/calling/entities/ECallMessage.js]: `z.calling.entities.ECallMessage`

File: app/script/calling/CallingService.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  constructor(client) {
+    this.logger = new z.util.Logger('z.calling.CallingService', z.config.LOGGER.OPTIONS);
+    this.client = client;
+    return this;
```
Comment (herrmannplatz) [inline app/script/calling/CallingService.js]: remove `return this`

File: app/script/calling/entities/ECallMessage.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  /**
+   * Create a session ID.
+   * @private
+   * @returns {string} Random four char session ID
+   */
+  _create_session_id() {
+    return new Array(E_CALL_MESSAGE_CONFIG.SESSION_ID_LENGTH)
+      .map(function() {
```
Comment (herrmannplatz) [inline app/script/calling/entities/ECallMessage.js]: single line?

File: app/script/calling/entities/EFlow.js [hunk 5]
```diff
@@ -0,0 +1,1361 @@
...
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.calling = z.calling || {};
+window.z.calling.entities = z.calling.entities || {};
+
+const E_FLOW_CONFIG = {
+  DATA_CHANNEL_LABEL: 'calling-3.0',
```
Comment (herrmannplatz) [inline app/script/calling/entities/EFlow.js]: these const leak into the global space. so either add a closure or make them static

File: app/script/calling/v3/CallCenter.js [hunk 6]
```diff
@@ -1117,8 +1117,8 @@ z.calling.v3.CallCenter = class CallCenter {
    * @returns {undefined} No return value
    */
   set_logging(is_enabled) {
-    this.logger.debug(`Set logging for webRTC Adapter: ${is_enabled}`);
     if (adapter) {
+      this.logger.debug(`Set logging for webRTC Adapter: ${is_enabled}`);
```
Comment (bennycode) [inline app/script/calling/v3/CallCenter.js]: It's "WebRTC"

Comment (gregor) [reply app/script/calling/v3/CallCenter.js]: Thanks. Fixed

**Pull request merged**